product_pricelist_supplierinfo/views/product_supplierinfo_view.xml

```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="product_supplierinfo_tree_view" model="ir.ui.view">
+        <field name="name">product.supplierinfo.tree.view.pricelist</field>
+        <field name="model">product.supplierinfo</field>
+        <field name="inherit_id" ref="product.product_supplierinfo_tree_view"/>
+        <field name="arch" type="xml">
+            <field name="price" position="after">
+                <field name="sale_margin"
+                   groups="product_pricelist_supplierinfo.group_supplierinfo_pricelist_sale_margin"/>
```

Comment:
You can put the group at view level, so that is not loaded at all and you optimize a bit that part.